app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,21 @@
+class SessionsController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by email: params[:session][:email].downcase
+
+    if user&.authenticate params[:session][:password]
```

Comment:
`&.` có tác dụng gì em

Reply:
Theo e hiểu , "&." ở đây có tác dụng:  đưa ra exception nếu đó là một object false, trong khi nếu dùng try() thì sẽ trả về nil

Reply:
em tìm hiểu thêm về keyword này `Safe Navigation Operator`

Reply:
&. là cái toàn tử an toàn check nil (user&.authenticate là viết tắt của user && user.authenticate), nhưng &. chỉ có tác dụng với nil và không có tác dụng với false